libtransmission/rpcimpl.c

```diff
@@ -927,6 +940,26 @@ static char const* torrentGet(tr_session* session, tr_variant* args_in, tr_varia
 ****
 ***/
 
+static char const* setLabels(tr_torrent* tor, tr_variant* list)
+{
+    int i;
+    const char * str;
+    size_t str_len;
+    const int n = tr_variantListSize(list);
+    tr_ptrArrayForeach(&tor->labels, tr_free);
+    tr_ptrArrayClear(&tor->labels);
```

Comment:
Slightly more idiomatic way of doing this:

```c
tr_ptrArrayDestruct(&tor->labels, tr_free);
*tor->labels = TR_PTR_ARRAY_INIT;
```

This is how it's done in e.g. announcer-udp and peer-mgr.c